### ISSUE TITLE
save_motor_window development

### ISSUE THREAD

**Issue by Cathyhjj** [spc-group/haven#256]
Discuss:
- [ ] whether to add metadata to the saved motor positions rather than just names
- [ ] whether to add `delete_saved_motor_positions(uid)` functions to motor_position.py, is the beamline db also MongoDB? 

Other observed issue:
`motor = self.motor_box.current_component()`, even though it's empty, it tried to get 

> A
FakeRobot(prefix='255idAustin', name='A', read_attrs=['i', 'i.user_readback', 'i.user_setpoint', 'j', 'j.user_readback', 'j.user_setpoint', 'k', 'k.user_readback', 'k.user_setpoint', 'l', 'l.user_readback', 'l.user_setpoint', 'm', 'm.user_readback', 'm.user_setpoint', 'n', 'n.user_readback', 'n.user_setpoint', 'x', 'x.user_readback', 'x.user_setpoint', 'y', 'y.user_readback', 'y.user_setpoint', 'z', 'z.user_readback', 'z.user_setpoint', 'rx', 'rx.user_readback', 'rx.user_setpoint', 'ry', 'ry.user_readback', 'ry.user_setpoint', 'rz', 'rz.user_readback', 'rz.user_setpoint', 'gripper_rbv', 'gripper_val', 'samples', 'samples.sample8', 'samples.sample8.present', 'samples.sample8.empty', 'samples.sample8.x', 'samples.sample8.y', 'samples.sample8.z', 'samples.sample8.rx', 'samples.sample8.ry', 'samples.sample8.rz', 'samples.sample9', 'samples.sample9.present', 'samples.sample9.empty', 'samples.sample9.x', 'samples.sample9.y', 'samples.sample9.z', 'samples.sample9.rx', 'samples.sample9.ry', 'samples.sample9.rz', 'samples.sample10', 'samples.sample10.present', 'samples.sample10.empty', 'samples.sample10.x', 'samples.sample10.y', 'samples.sample10.z', 'samples.sample10.rx', 'samples.sample10.ry', 'samples.sample10.rz', 'samples.sample14', 'samples.sample14.present', 'samples.sample14.empty', 'samples.sample14.x', 'samples.sample14.y', 'samples.sample14.z', 'samples.sample14.rx', 'samples.sample14.ry', 'samples.sample14.rz', 'samples.sample15', 'samples.sample15.present', 'samples.sample15.empty', 'samples.sample15.x', 'samples.sample15.y', 'samples.sample15.z', 'samples.sample15.rx', 'samples.sample15.ry', 'samples.sample15.rz', 'samples.sample16', 'samples.sample16.present', 'samples.sample16.empty', 'samples.sample16.x', 'samples.sample16.y', 'samples.sample16.z', 'samples.sample16.rx', 'samples.sample16.ry', 'samples.sample16.rz', 'samples.sample20', 'samples.sample20.present', 'samples.sample20.empty', 'samples.sample20.x', 'samples.sample20.y', 'samples.sample20.z', 'samples.sample20.rx', 'samples.sample20.ry', 'samples.sample20.rz', 'samples.sample21', 'samples.sample21.present', 'samples.sample21.empty', 'samples.sample21.x', 'samples.sample21.y', 'samples.sample21.z', 'samples.sample21.rx', 'samples.sample21.ry', 'samples.sample21.rz', 'samples.sample22', 'samples.sample22.present', 'samples.sample22.empty', 'samples.sample22.x', 'samples.sample22.y', 'samples.sample22.z', 'samples.sample22.rx', 'samples.sample22.ry', 'samples.sample22.rz'], configuration_attrs=['i', 'i.user_offset', 'i.user_offset_dir', 'i.velocity', 'i.acceleration', 'i.motor_egu', 'j', 'j.user_offset', 'j.user_offset_dir', 'j.velocity', 'j.acceleration', 'j.motor_egu', 'k', 'k.user_offset', 'k.user_offset_dir', 'k.velocity', 'k.acceleration', 'k.motor_egu', 'l', 'l.user_offset', 'l.user_offset_dir', 'l.velocity', 'l.acceleration', 'l.motor_egu', 'm', 'm.user_offset', 'm.user_offset_dir', 'm.velocity', 'm.acceleration', 'm.motor_egu', 'n', 'n.user_offset', 'n.user_offset_dir', 'n.velocity', 'n.acceleration', 'n.motor_egu', 'x', 'x.user_offset', 'x.user_offset_dir', 'x.velocity', 'x.acceleration', 'x.motor_egu', 'y', 'y.user_offset', 'y.user_offset_dir', 'y.velocity', 'y.acceleration', 'y.motor_egu', 'z', 'z.user_offset', 'z.user_offset_dir', 'z.velocity', 'z.acceleration', 'z.motor_egu', 'rx', 'rx.user_offset', 'rx.user_offset_dir', 'rx.velocity', 'rx.acceleration', 'rx.motor_egu', 'ry', 'ry.user_offset', 'ry.user_offset_dir', 'ry.velocity', 'ry.acceleration', 'ry.motor_egu', 'rz', 'rz.user_offset', 'rz.user_offset_dir', 'rz.velocity', 'rz.acceleration', 'rz.motor_egu', 'acc', 'vel', 'remote_control', 'program_rbv', 'installation', 'close_safety_popup', 'unlock_protective_stop', 'restart_safety', 'program_running', 'safety_status', 'power_rbv', 'gripper_activated', 'gripper_force', 'current_sample', 'home', 'cal_stage', 'samples', 'samples.sample8', 'samples.sample9', 'samples.sample10', 'samples.sample14', 'samples.sample15', 'samples.sample16', 'samples.sample20', 'samples.sample21', 'samples.sample22'])

`

**Comment by canismarko** [comment]
> is the beamline db also MongoDB?

Yes, both the motor positions and our scans get saved to the same mongo database, but in different collections.

Now that I think about it, maybe we should scrap the saved motor position collection and just do a ``bp.count`` on the motors to save their current positions.

It would mean that saving positions has to happen in a run engine, but it would open up some advantages like API access and being able to export them to a csv, etc.

Thoughts?

**Comment by Cathyhjj** [comment]
Status 
- [x]  add a checkbox in front of the dates
- [x]  merge refreshing and filter button
- [x]  put a splitter to resize the component selectors and the save position TableWidget
- [ ]  Save queue now or later()
- [ ]  Recall now or recall later()

**Comment by Cathyhjj** [comment]
@canismarko 
Maybe: add name contains filter in `get_motor_position`

**Comment by Cathyhjj** [comment]
bug, connect to PyDM signals only when the combobox edit is finished

**Comment by Cathyhjj** [comment]
2024-10-15 

- [] try and except block for user_readback control
- [] try and except for the comboBox LineEdit
- [] When initializing the window, there's always one more entry
- [] They all should be queue buttons
- [] put def `update_queue_status(self, status)`


- Wait until Mark finishes main changes on Ion chambers

**Comment by Cathyhjj** [comment]
- [ ] work on merging main into save_motor_window

**Comment by Cathyhjj** [comment]
issues

- [ ] When initializing the window, there's always one more entry
- [ ] Refresh positions do not work
- [ ] use contains instead Regex, which doesn't work yet (maybe in the future)
- [ ] remove redundent refresh_position_list
